test/unit/toggle/toggle-component.js

```diff
@@ -0,0 +1,147 @@
+import Toggle from '../../../src/components/toggle';
+import ToggleView from '../../../src/views/toggle';
+
+describe('Toggle Component class', () => {
+  let toggle;
+  let cart;
+  let node;
+
+  beforeEach(() => {
+    node = document.createElement('div');
+  });
+
+  describe('constructor', () => {
+    beforeEach(() => {
+      cart = {
+        node: document.createElement('div'),
+      };
+      document.body.appendChild(cart.node);
+    });
+
+    afterEach(() => {
+      document.body.removeChild(cart.node);
+    });
+
+    it('sets typeKey to toggle', () => {
+      toggle = new Toggle({node}, {cart});
+      assert.equal(toggle.typeKey, 'toggle');
+    });
+
+    it('sets node to node in config if provided', () => {
+      toggle = new Toggle({node}, {cart});
+      assert.equal(toggle.node, node);
+    });
+
+    it('sets node to a div element before cart node if config does not provide node', () => {
+      const div = document.createElement('div');
+      const insertBeforeStub = sinon.stub(Node.prototype, 'insertBefore').returns(node);
+      const createElementStub = sinon.stub(document, 'createElement').returns(div);
+
+      toggle = new Toggle({}, {cart});
+      assert.calledOnce(insertBeforeStub);
+      assert.calledWith(insertBeforeStub, div, cart.node);
+      assert.equal(toggle.node, node);
+
+      insertBeforeStub.restore();
+      createElementStub.restore();
+    });
+
+    it('creates a new toggle view', () => {
+      toggle = new Toggle({node}, {cart});
+      assert.instanceOf(toggle.view, ToggleView);
+    });
+  });
+
+  describe('prototype methods', () => {
+    let toggleVisibilitySpy;
+
+    beforeEach(() => {
+      toggleVisibilitySpy = sinon.spy();
+      cart = {
+        node: document.createElement('div'),
```

Comment:
Similar to above, I think we can remove this as well as the `appendChild` and `removeChild`, since we're passing in the node.

Reply:
I think we could also remove the node from the cart, since it won't be used